src/test/compile-fail/issue-13352.rs

```diff
@@ -23,4 +23,6 @@ fn main() {
         unsafe { libc::exit(0 as libc::c_int); }
     });
     2_usize + (loop {});
+    //~^ ERROR E0277
```

Comment:
(of course it might be even nicer to not require the type annotation at all, since after all the addition is an unreachable expression in the control flow.  but I'd be satisfied with an annotation rather than trying to go further.)